### PR TITLE
Fix Social Security Calibration

### PR DESCRIPTION
**Pull request by jdebacker**
This PR fixes the calibration of the `theta` parameter, which determines the social security replacement rate.

@rickecon and @kerkphil - can you take a look at the `tax.replacement_rate_vals()` function to see if this approach to calibrating `theta` seems right?

This PR is a work in progress because with the current calibration of `theta`, the model is able to find a solution for the case of per period balanced budgets, but can only solve of the SS (not the TP) when the budget is unbalanced.  I'm not sure of why this is, but the TPI cycles and does not converge.

### CODE REVIEW

**Comment by kerkphil** [comment]
@jdebacker , @rickecon - the code looks good to me.  I'm trying to remember what the replacement rate is conceptually in the model.  From the code we have: PIA = theta * AIME / LifetimeEffectiveLabor.  Why do we need the lifetime effective labor adjustment?

**Comment by jdebacker** [comment]
@kerkphil I think the main change in the way `theta` is calculated now is that the AIME is based on the last 35 years of earnings.  But there is also a change in how the PIA `theta` are computed that better follows my understanding of the law.  In particular, given that the PIA is the insurance amount, this line of code:

`theta = PIA * (e * nssmat).mean(0) / AIME`

Didn't make sense to me.   Currently we are doing:

`theta = (PIA*(12.0*S/80)) / (factor_ss*wss)`

Which taxes the PIA, turns it from a monthly amount to an annual amount.  It then transforms it from dollars to model units via the division by the `factor`.  The wage rate in the denominator is because benefits adjusted by wage growth, so we scale by wages.  When we put benefits into the budget of model households, we do `theta*w`.

**Comment by kerkphil** [comment]
@jdebacker OK, I looked at the old code.  Let me look at the PR code, but your explanation of theta makes a lot more sense to me.

**Comment by kerkphil** [comment]
@jdebacker Yes, this code looks better to me.  One issue, is AIME based on the past 35 years of wages or the 35*12 highest months of adjusted wages over the workers whole lifetime?  My impression was the latter.

**Comment by jdebacker** [comment]
@kerkphil - yes, you are right that it's the highest 35 years.  Thinking more about it - that's something we should be able to do.  Let me update the PR to reflect that.

**Comment by kerkphil** [comment]
@jdebacker Since we are looking at the steady state values, the highest 35 years of wages are probably the 35 years immediately before retirement anyway, so it may not make that much difference.  However, if it's not too difficult it would be best to recode, just to be safe.

File: Python/ogusa/tax.py [hunk 1]
```diff
@@ -35,41 +35,27 @@ def replacement_rate_vals(nssmat, wss, factor_ss, params):
...
-            PIA = 1879.86 + .15 * (AIME - 4517.0)
-        theta = PIA * (e * nssmat).mean(0) / AIME
-        # Set the maximum replacment rate to be $30,000
-        maxpayment = 30000.0 / (factor_ss * wss)
-        if theta > maxpayment:
-            theta = maxpayment
-    theta = 0  # setting theta = 0 since we are including social security income in capital income (CHECK)
+            PIA[j] = 1879.86 + .15 * (AIME[j] - 4517.0)
+    # Set the maximum replacment rate to be $30,000
+    maxpayment = 3501.00 #30000.0/12.0
```
Comment (jdebacker) [inline Python/ogusa/tax.py]: @kerkphil I updated the AIME calculation to take the highest 35 years of earnings.  I also changed the PIA max amount.  $30,000 was an annual amount, I think.  Replaced with $3501 per month.  Not 100% I got this right.  Took highest number from 2015 from [this table](https://www.ssa.gov/oact/cola/examplemax.html).

Comment (kerkphil) [inline Python/ogusa/tax.py]: Looks great!

**Comment by jdebacker** [comment]
TPI not seems to converge (not cycling).  Two updates helped with this:
1) Better initial guesses
2) A smoother transition path for government spending.

**Comment by jdebacker** [comment]
Fiscal parameters are now updated per discussion in Issue #254.  

Removing WIP.

**Comment by jdebacker** [comment]
Merging.  Will address further fiscal calibration issues in another PR.

**Pull request merged**